### PR TITLE
Issue666 wav read

### LINKED ISSUE #666: Migrate g.wavread from GGIR to GGIRread
So far I have focussed on g.cwaread and g.binread, but this can also be done for g.wavread.
That may then also be a good opportunity to test whether it still works and tidy up the code where needed.

### PR DESCRIPTION
**Pull request by jhmigueles**
Fixes #666 
`g.wavread` now deprecated and migrated to the GGIRread package (readWav). 

**IMPORTANT:** I think that for this to work, we need first to get the [GGIRread PR #24](https://github.com/wadpac/GGIRread/pull/24) merged and into CRAN.

Checklist before merging:

- [ ] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [ ] Updated or expanded the documentation.
- [ ] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` and `CITATION.cff` files.

### CODE REVIEW

File: man/g.wavread.Rd [hunk 1]
```diff
@@ -1,39 +1,16 @@
 \name{g.wavread}
 \alias{g.wavread}
 \title{
-function to read .wav files as produced by the accelerometer named 'Axivity'
-
+  Deprecated, internally replaced by function readAxivity from the GGIRread package
```
Comment (vincentvanhees) [inline man/g.wavread.Rd:4]: `replaced by function readAxivity` should read `replaced by function readWav`

**Review by vincentvanhees — COMMENTED** [review]
Thanks, looks good, one minor language issue.

**Comment by jhmigueles** [comment]
Just seen that in inst/testfiles there is a wav file. Was this only used for the g.wavread function test? If so, we can remove that test file before merging.

**Comment by vincentvanhees** [comment]
> Just seen that in inst/testfiles there is a wav file. Was this only used for the g.wavread function test? If so, we can remove that test file before merging.

yes, it can be removed

**Comment by vincentvanhees** [comment]
Looks good. I suppose it is now important that I first make a new GGIRread release on CRAN before merging these update to GGIR.

**Comment by vincentvanhees** [comment]
Update: CRAN just accepted the new release for GGIRread, which means that later this week when it is online we can merge this PR.

**Comment by vincentvanhees** [comment]
The g.inspectfile code is still relying on tuneR, by which GGIR will struggle to extract the sampling frequency because tuneR is no longer a dependency of GGIR. I am now fixing this.

**Comment by vincentvanhees** [comment]
Hi @jhmigueles I am now having the following problem:

When I run test_greadaccfile on its on, the test passes, but when I run the entire package via toolbar Build => Test package. I get an error for the wav file reading: 

image

The main changes I made are:
- wav testfile removed and the unit test now relies on the test file from GGIR.
- tuneR dependency removed from g.inspectfile.

Could you check whether you get the same error? If not then maybe this is a local problem for me. I am waiting for the GitHub Actions tests to run as they may also be able to clarify whether this is a local problem or not.

**Comment by vincentvanhees** [comment]
>  I cannot see either where the serial number info is in the wav file header

GGIR::g.extractheadervars() does this for GGIR, because serial number is not consistently in the same location in wav file headers. 

> Proposed change in test_greadaccfile --> expect_equal(EHV$deviceSerialNumber, "not extracted")

How did you arrive at this conclusion? As far as I can see the error originates when readWav() is called during the package check and is not related to the unit test itself. I do not get the error when I run test_greadaccfile on its own or when I use readWav() directly on the test wavfile, but I do get the error when I run the RStudio package tests. Do you experience the same?

I think it is best to first make sure that everything works in this branch. Once this branch is merged it will become a lot harder to make changes to function readWav, because from then onward we always need to facilitate both new and old versions of the GGIRread code in GGIR, because we cannot update both packages at the same time on CRAN.

Some additional reflections:
- I see you removed a large chunk of code from the `dformat == 3` section in function g.inspectfile but did not migrate it to GGIRread. It may be worth checking that this code is truly redundant.
- Maybe things went wrong when I removed tuneR dependency from this branch. Ideally tuneR should be a dependency of GGIRread only and not of GGIR any longer.
- It may be easiest to first aim for a working development branch in GGIRread and a development branch in GGIR that are compatible before making any more package releases with updates concerning wav.
- If all of this takes too much effort another solution is to entirely deprecate the wav functionality, and bring it back to life again if anyone is willing to pay us for the development work. The biggest challenge with the wav read are the inconsistencies in wav file header formats across wav-format generations.

**Comment by vincentvanhees** [comment]
Also, I think it is probably best to migrate more wav-related functionality to GGIRread, such that the code there is fully responsible for extracting serial number, sample frequency. At the moment, the functionality is possibly still to much split between the packages.

**Comment by jhmigueles** [comment]
> I do not get the error when I run test_greadaccfile on its own or when I use readWav() directly on the test wavfile, but I do get the error when I run the RStudio package tests. Do you experience the same?

In my case it is the opposite, the tests passes through teh RStudio package tests, but it does not when I only run the test_greadaccfile:
image

Today I have been trying to get the serial number of the wav file in testfiles using an older GGIR version (1.7-0), but I always get "not extracted". I'm not sure what the problem is... Let me know what you think is best.

**Comment by vincentvanhees** [comment]
Closing this PR now as we are not actively working on it.
First step will be to update to GGIRread to automatically extract the wav file header and handle the variations in its formatting.

**Pull request closed without merging**